packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart

```diff
@@ -223,7 +223,22 @@ class FirebaseOptions {
 
   @override
   int get hashCode {
-    return hashObjects(asMap.entries);
+    return hashValues(
+      apiKey ?? googleAppID,
```

Comment:
THe `googleAppId` here should be on the line below?

Reply:
@Ehesp 

Is this exiting code also mistake?
If so, I'll correct both.
https://github.com/FirebaseExtended/flutterfire/blob/2273334b4d661850f5120b5b3c96bb617a964d05/packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart#L199

Reply:
Hmm potentially yeah, I'll check it out.

Reply:
According to the definition, the `googleAppID` is deprecated and replaced by `apiKey` which is why it's written this way.

https://github.com/FirebaseExtended/flutterfire/blob/2273334b4d661850f5120b5b3c96bb617a964d05/packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart#L52

Reply:
Definitely seems like it, good catch.

https://github.com/FirebaseExtended/flutterfire/blob/2273334b4d661850f5120b5b3c96bb617a964d05/packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart#L80
https://github.com/FirebaseExtended/flutterfire/blob/2273334b4d661850f5120b5b3c96bb617a964d05/packages/firebase_core/firebase_core_platform_interface/lib/src/firebase_options.dart#L81

Reply:
Yah, could you fix that apart of this PR?

Reply:
@Ehesp 
Okay, I fixed them at https://github.com/FirebaseExtended/flutterfire/pull/3263/commits/def21881b36563598f40e6e509d5e89395c89e43 👍